File: framec_tests/golang/simple_handler_calls/simple_handler_calls_test.go
package simple_handler_calls

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

/* Test a basic handler call. */
func TestSimpleCall(t *testing.T) {
	sm := NewSimpleHandlerCalls()
	x := sm.(*simpleHandlerCallsStruct)
	sm.C()
	assert.Equal(t, SimpleHandlerCallsState_A, x._compartment_.State)
}

/* Test that a handler call terminates the current handler. */
func TestCallsTerminateHandler(t *testing.T) {
	sm := NewSimpleHandlerCalls()
	x := sm.(*simpleHandlerCallsStruct)
	sm.D()
	assert.Equal(t, SimpleHandlerCallsState_B, x._compartment_.State)

	sm = NewSimpleHandlerCalls()
	x = sm.(*simpleHandlerCallsStruct)
	sm.E()
	assert.Equal(t, SimpleHandlerCallsState_B, x._compartment_.State)

}

File: framec_tests/java/Simple_handler_calls/Simple_handler_calls_test.java
package framec_tests.java.Simple_handler_calls;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

class SimpleHandlerCallsController extends SimpleHandlerCalls {

    SimpleHandlerCallsController() {
      super();
    }
}

public class Simple_handler_calls_test {

    /* Test a basic handler call. */
    @Test
    public void TestSimpleCall(){
        SimpleHandlerCallsController sm = new SimpleHandlerCallsController();
        sm.C();
        assertEquals(sm.state_info(), "1");
    }

    /* Test that a handler call terminates the current handler. */
    @Test
    public void TestCallsTerminateHandler(){
        SimpleHandlerCallsController sm  = new SimpleHandlerCallsController();
        sm.D();
        assertEquals(sm.state_info(), "2");

        sm = new SimpleHandlerCallsController();
        sm.E();
        assertEquals(sm.state_info(), "2");
    }
    
}

